File: app/src/main/java/com/grsu/guideapp/holders/types/TypeHolder.java
package com.grsu.guideapp.holders.types;

import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;
import com.grsu.guideapp.R;
import com.grsu.guideapp.base.BaseViewHolder;
import com.grsu.guideapp.base.listeners.ItemClickListener;
import com.grsu.guideapp.models.DtoType;

public class TypeHolder extends BaseViewHolder<DtoType> {

//    @BindView(R.id.item_type_image)
    ImageView mItemTypeImage;
//    @BindView(R.id.item_type_name)
    TextView mItemTypeName;

    public TypeHolder(View itemView, ItemClickListener listener) {
        super(itemView, listener);
    }

    @Override
    public void bind(DtoType type) {
        mItemTypeName.setText(type.getName());
        mItemTypeImage.setImageResource(getImage(type.getId()));
    }

    private int getImage(int idCategory) {
        return R.mipmap.ic_launcher;
    }
}

File: app/src/main/java/com/grsu/guideapp/holders/types/TypeHolder.kt
package com.grsu.guideapp.holders.types

import android.view.View
import android.widget.ImageView
import android.widget.TextView
import com.grsu.guideapp.R
import com.grsu.guideapp.base.BaseViewHolder
import com.grsu.guideapp.database.vo.TypeItemVO
import com.grsu.guideapp.utils.extensions.toBitmap

class TypeHolder(
    itemView: View,
    listener: ((View, Int) -> Unit)
) : BaseViewHolder<TypeItemVO>(itemView, listener) {

    private val name = itemView.findViewById(R.id.item_type_name) as? TextView
    private val image = itemView.findViewById(R.id.item_type_image) as? ImageView

    override fun bind(item: TypeItemVO): Unit = item.let {
        name?.text = it.name
        val bytes = it.image
        if (bytes.isEmpty()) image?.setImageResource(R.drawable.ic_launcher_foreground)
        else image?.setImageBitmap(it.image.toBitmap())
    }
}